### PR TITLE
[TTAHUB-559] Update creator and collaborator data on legacy reports

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Some legacy reports are missing creator or collaborator data in the database despite having that data present in the smartsheet data column. This script attempts to assign collaborators from the "otherSpecialists" field and creators from the "createdBy" field.

## How to test
Run the script. You may not have data in your DB that works (emails in the "imported" data in my database don't match any of the users), so you might have to create some.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-559

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/tools/updateLegacyCreatorsAndCollaborators.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        through: {
+          attributes: [],
+        },
+        required: false,
+      },
+    ],
+  });
+
+  // so here is where the filtering happens
+  const reports = rawReportData.filter((r) => r.collaborators.length === 0 || r.userId === null);
```
Comment (jasalisbury) [inline src/tools/updateLegacyCreatorsAndCollaborators.js:112]: Are we worried about a situation where a report has some collaborators but not all?

File: src/tools/updateLegacyCreatorsAndCollaborators.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    } else {
+      logger.info('No matching creator found');
+    }
+  }
+
+  return sequelize.transaction(async (transaction) => {
+    await Promise.all([
+      ...updatedCollaborators.map((c) => ActivityReportCollaborator.create({
+        activityReportId: id,
+        userId: c.id,
```
Comment (jasalisbury) [inline src/tools/updateLegacyCreatorsAndCollaborators.js:74]: If we need to handle a report that has some collaborators but not all we will have to make sure we don't dup `ActivityReportCollaborators`

**Comment by thewatermethod** [comment]
Based on a conversation with Angela today and some concerns about trying to parse this data programmatically with a script, we've scaled back the expectation here. Basically - if there are emails in those fields, they should be added to the appropriate places. 

Not accounting for any of the weirdness around names, etc, at this point. That will likely either have to be a separate script or something that is addressed manually, either through the new UI that's been discussed a little, or through a situation where the a CSV is generated with all the needed ids and then run through a script.

File: package.json [hunk 3]
```diff
@@ -71,23 +71,9 @@
...
     "changeReportStatus": "node ./build/server/tools/changeReportStatusCLI.js",
     "changeReportStatus:local": "./node_modules/.bin/babel-node ./src/tools/changeReportStatusCLI.js",
-    "updateLegacyPopulations": "node ./build/server/tools/updateLegacyPopulationsCLI.js",
-    "updateLegacyPopulations:local": "./node_modules/.bin/babel-node ./src/tools/uupdateLegacyParticipantsCLI.js",
-    "updateLegacyParticipants": "node ./build/server/tools/updateLegacyPopulationsCLI.js",
-    "updateLegacyParticipants:local": "./node_modules/.bin/babel-node ./src/tools/updateLegacyParticipantsCLI.js",
-    "importActivityReportDuration": "node ./build/server/tools/importActivityReportDuration.js",
-    "importActivityReportDuration:local": "./node_modules/.bin/babel-node ./src/tools/importActivityReportDuration.js",
-    "updateTopicNames": "node ./build/server/tools/updateTopicNamesCLI.js",
-    "updateTopicNames:local": "./node_modules/.bin/babel-node ./src/tools/updateTopicNamesCLI.js",
```
Comment (jasalisbury) [inline package.json]: :taco:

**Review by jasalisbury — APPROVED** [review]

**Pull request merged**